cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/Language.kt

```diff
@@ -58,12 +65,84 @@ abstract class Language<T : LanguageFrontend> : Node() {
     open val primitiveTypes: Set<String>
         get() = setOf("byte", "short", "int", "long", "float", "double", "boolean", "char")
 
+    /** The access modifiers of this programming language */
+    open val accessModifiers: Set<String>
+        get() = setOf("public", "protected", "private")
+
     /** Creates a new [LanguageFrontend] object to parse the language. */
     abstract fun newFrontend(
         config: TranslationConfiguration,
         scopeManager: ScopeManager = ScopeManager()
     ): T
 
+    open fun getTypeOf(typeString: String, modifier: ObjectType.Modifier): Type? {
+        return when (typeString) {
+            "byte" -> IntegerType(typeString, 8, this, modifier)
+            "short" -> IntegerType(typeString, 16, this, modifier)
+            "int" -> IntegerType(typeString, 32, this, modifier)
+            "long" -> IntegerType(typeString, 64, this, modifier)
+            "long long int" -> IntegerType(typeString, 64, this, modifier)
+            "float" -> FloatingPointType(typeString, 32, this, modifier)
+            "double" -> FloatingPointType(typeString, 64, this, modifier)
+            "boolean" -> IntegerType(typeString, 1, this, modifier)
+            "char" -> IntegerType(typeString, 8, this, ObjectType.Modifier.NOT_APPLICABLE)
+            else -> null
+        }
+    }
+
+    // TODO: These are language specific too.
+    private val VOLATILE_QUALIFIER = "volatile"
+    private val FINAL_QUALIFIER = "final"
+    private val CONST_QUALIFIER = "const"
+    private val RESTRICT_QUALIFIER = "restrict"
+    private val ATOMIC_QUALIFIER = "atomic"
+
+    open fun updateQualifier(qualifierString: String, old: Qualifier): Boolean {
+        if (this !is HasQualifier || qualifierString !in qualifiers) {
+            return false
+        }
+        val qualifier = old
+        when (qualifierString) {
+            FINAL_QUALIFIER,
```

Comment:
Do we even use these qualifiers?

Reply:
What do you mean with "use"? We don't know what the end-user does with this information. But I agree, it seems to be information which is not too interesting